### PR TITLE
[CMSP-992] UI changes

### PR DESCRIPTION
**Pull request by jazzsequence**
This PR:

* updates some of the language in the notices and alerts
* adds a new build pipeline using Gulp and Sass (and removes old NPM dependencies that weren't being used)
* adds `build_node_assets` to the build-tag-release workflow
* adds styling for Pantheon Page Cache Max Age field (uses selectors added in https://github.com/pantheon-systems/pantheon-mu-plugin/pull/46)
* adds new elements and overrides default HTML elements (based on hooks added in https://github.com/pantheon-systems/pantheon-mu-plugin/pull/46)
* adds styles (and a filter that is commented-out) for an info bar in the Max Age setting per the Figma designs (disabled because we aren't using it)
* adds other styles and text per the Figma designs
* changes the input box to a select (dropdown) if the value is not filtered and not manually set. (a warning is displayed if a custom value is saved that is not one of the recommended (default) options
* fixes some logic around warnings and alerts
* adds tests for new components

## Demo
https://www.loom.com/share/686250ad45b3493fa5f8f250cf233345

## Screenshots
**Global notice when max age of 600 is reset to the default**
Screenshot 2024-05-20 at 2 37 35 PM

**Pantheon Page Cache warning when max age is _far_ less than the recommended value** (Note: `max-age` is now `max age` in the code)
Screenshot 2024-05-20 at 2 37 28 PM

**Site Health recommendation when max age is _far_ less than the recommended value** (Note: all instances of `max-age`/`Max-Age` have been replaced with their equivalent without the `-` in the code)
Screenshot 2024-05-21 at 9 14 44 AM

**Pantheon Page Cache warning when max age is closer, but still less than, the recommended value** (Note: `max-age` is now `max age` in the code)
Screenshot 2024-05-20 at 2 37 07 PM

**Site Health recommendation when max age is closer, but still less than, the recommended value** (Note: all instances of `max-age`/`Max-Age` have been replaced with their equivalent without the `-` in the code)
Screenshot 2024-05-21 at 9 11 43 AM

**Pantheon Page Cache options page with new settings**
Screenshot 2024-05-21 at 9 09 00 AM

**Pantheon Page Cache Max Age dropdown**
Screenshot 2024-05-21 at 9 09 15 AM

**Pantheon Page Cache Max Age with custom value** (Note: This would only occur if a value other than 600 and any of the default options had been set previously.)
Screenshot 2024-05-21 at 1 59 30 PM

**Pantheon Page Cache Max Age with _filtered_ value** (Note: If a custom/different value is saved in the database, the filtered value replaces it in the (disabled) input)
Screenshot 2024-05-21 at 3 28 41 PM

**Pantheon Cache Max Age with info bar** (Note: This is controlled via a filter that can be hooked to `__return_true`/`__return_false` to toggle. By default it is not displayed. Styles are included in this PR per the Figma and in case we choose to use it in the future.
Screenshot 2024-05-21 at 3 30 20 PM

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -19,9 +19,9 @@
 /composer.json export-ignore
 /composer.lock export-ignore
 /CONTRIBUTING.md export-ignore
-/Gruntfile.js export-ignore
+/gulpfile.mjs export-ignore
```
Comment (pwtyler) [inline .gitattributes:22]: confirming, this should be a replacement, not an adendum?

**Comment by scottbuscemi** [comment]
Screenshot 2024-05-21 at 3 00 43 PM

Looks like "Recommended Setting" is using the hardcoded value rather than just saying '1 day'

**Comment by jazzsequence** [comment]
@scottbuscemi 
> Looks like "Recommended Setting" is using the hardcoded value rather than just saying '1 day'

Explained in the demo. If a value is filtered, the filtered value takes the place of the recommendation/becomes the new recommendation. If this is different than what you would like, we can change it, but the assumption is, if you're filtering it, you're breaking the seal on the warranty.

**Comment by scottbuscemi** [comment]
Let's remove the "Value range" sentence

**Comment by scottbuscemi** [comment]
If someone is doing their own thing, we can just not recommend something anymore :)

**Comment by jazzsequence** [comment]
> If someone is doing their own thing, we can just not recommend something anymore :)

Sure can.

File: inc/admin-interface.php [hunk 2]
```diff
@@ -36,6 +35,149 @@ function bootstrap() {
...
+function enqueue_admin_assets() {
+	$screen = get_current_screen();
+
+	if ( 'settings_page_pantheon-cache' !== $screen->id ) {
+		return;
+	}
+
+	$debug = defined( 'WP_DEBUG' ) && WP_DEBUG ? '-' . time() : '';
+
+	wp_enqueue_style( 'papc-admin', plugin_dir_url( __DIR__ ) . 'assets/css/styles.css', [], '2.0.0' . $debug ); // If WP_DEBUG is true, append a timestamp to the end of the path so we get a fresh copy of the css.
```
Comment (pwtyler) [inline inc/admin-interface.php]:
```suggestion
	// If WP_DEBUG is true, append a timestamp to the end of the path so we get a fresh copy of the css.
	$debug = defined( 'WP_DEBUG' ) && WP_DEBUG ? '-' . time() : '';
	wp_enqueue_style( 'papc-admin', plugin_dir_url( __DIR__ ) . 'assets/css/styles.css', [], '2.0.0' . $debug );
```

**Comment by jazzsequence** [comment]
Screenshot 2024-05-21 at 4 07 26 PM

@scottbuscemi

File: inc/admin-interface.php [hunk 3]
```diff
@@ -36,6 +35,149 @@ function bootstrap() {
...
+}
+
+/**
+ * Add a header to the max-age setting field.
+ */
```
Comment (pwtyler) [inline inc/admin-interface.php]: missing since and return

File: inc/admin-interface.php [hunk 4]
```diff
@@ -36,6 +35,149 @@ function bootstrap() {
...
+		}
+		?>
+		<span class="pantheon-cache-default-max-age-info-bar"><i class="dashicons dashicons-info"></i><?php echo wp_kses_post( 'Boost site speed with a higher GCDN cache max age.', 'pantheon-advanced-page-cache' ); ?></span>
+	<?php
+	return ob_get_clean();
+}
+
+/**
+ * Add a description to the max-age setting field.
+ */
```
Comment (pwtyler) [inline inc/admin-interface.php]: since and return 🔢

**Review by pwtyler — COMMENTED** [review]
A few comments. have not watched recording yet.

File: .gitattributes (hunk 1 above)
Comment (jazzsequence) [reply .gitattributes:22]: Correct. Using Gulp _instead of_ Grunt (which we weren't really using for anything anyway).

**Comment by jazzsequence** [comment]
Cache max age recommendation message updates:

**Current value is _above_ the recommendation**
Screenshot 2024-05-22 at 9 05 20 AM

**Current value is set to the recommendation**
Screenshot 2024-05-22 at 9 05 38 AM

**Current value is _below_ the recommendation**
Screenshot 2024-05-22 at 9 06 08 AM

**Current value is _filtered below_ the recommendation**
Screenshot 2024-05-22 at 9 06 37 AM

**Current value is _filtered above_ the recommendation**
Screenshot 2024-05-22 at 9 07 05 AM

**Note:** Updated to remove duplicative "current" in the message(s).

File: inc/admin-interface.php [hunk 5]
```diff
@@ -36,6 +35,160 @@ function bootstrap() {
...
+	$recommended_message = __( 'Your cache maximum age is currently set to the recommended value.', 'pantheon-advanced-page-cache' );
+	$recommendation_message = get_current_max_age() > WEEK_IN_SECONDS ? $above_recommended_message : ( get_current_max_age() < WEEK_IN_SECONDS ? $below_recommended_message : $recommended_message );
+	$filtered_message = $filtered ? sprintf(
+		// translators: %s is the humanized max-age.
+		__( 'This value has been hardcoded to %s via a filter.', 'pantheon-advanced-page-cache' ),
+		'<strong>' . humanized_max_age() . '</strong>'
+	) : '';
+	$pantheon_cache = get_option( 'pantheon-cache', [] );
+	$custom_ttl = isset( $pantheon_cache['default_ttl'] ) && ! array_key_exists( $pantheon_cache['default_ttl'], max_age_options() );
+	$filtered_message .= $custom_ttl && ! $filtered ? '<br />' . __( '<strong>Warning:</strong>The cache max age is not one of the recommended values. If this is not intentional, you should remove this custom value and save the settings, then select one of the options from the dropdown.', 'pantheon-advanced-page-cache' ) : '';
```
Comment (pwtyler) [inline inc/admin-interface.php]: This is dense to read and parse out reading code but not going to block on it. For example, if $custom_ttl is a bool, the variable name should read as one.

https://github.com/pantheon-systems/Standards/blob/b0a15d37e61461226f1d6061bb3ac5d087548948/Data_Standards.md?plain=1#L15-L16

I know you love ternary operators but it's a pain long-term.

File: inc/admin-interface.php [hunk 6]
```diff
@@ -36,6 +35,160 @@ function bootstrap() {
...
+ * @return int
+ */
+function update_default_ttl_input( $default_input ) {
+	$slug = 'pantheon-cache';
+	$pantheon_cache = get_option( $slug, [] );
+	$default_ttl = isset( $pantheon_cache['default_ttl'] ) && $pantheon_cache['default_ttl'] !== 0 ? $pantheon_cache['default_ttl'] : WEEK_IN_SECONDS;
+	$options = max_age_options();
+	$filtered = has_filter( 'pantheon_cache_default_max_age' );
+	$custom = ! array_key_exists( $default_ttl, $options ) && ! $filtered;
+	$output = '';
```
Comment (pwtyler) [inline inc/admin-interface.php]: less dense, but same thoughts re: bool variables as above.

**Review by pwtyler — APPROVED** [review]
I would love some cleanup and clarity as described in a couple places, but will leave at your discretion.

**Pull request merged**